pkg/segment/context/context.go

```diff
@@ -111,6 +114,22 @@ func SetInteractive(ctx context.Context, interactive bool) {
 	setContextProperty(ctx, InteractiveMode, interactive)
 }
 
+// SetFlags sets flags property for telemetry to record what flags were used
+func SetFlags(ctx context.Context, flags *pflag.FlagSet) {
+	var changedFlags []string
+	flags.VisitAll(func(f *pflag.Flag) {
+		if f.Changed {
+			if f.Name == "logtostderr" {
+				// skip "logtostderr" flag, for some reason it is showing as changed even when it is not
+				return
+			}
+			changedFlags = append(changedFlags, f.Name)
+		}
+	})
+	// the flags can't have spaces, so the output is space separated list of the flag names
+	setContextProperty(ctx, Flags, strings.Join(changedFlags, " "))
```

Comment:
Just curious, because I don't know yet what this would give in Segment: does the value need to be a string? The signature of `setContextProperty` allows to pass an `interface{}`, so it might be better to just pass the `changedFlags` slice as is, if possible?
When testing locally (using `ODO_DEBUG_TELEMETRY_FILE`), I found the resulting JSON to be better, but I am not sure if this translates into usable data in Segment:

```json
{
  "event": "odo init",
  "properties": {
    "duration": 1804,
    "error": "starter project \"go\" not found in devfile\nthe command failed, the devfile has been removed from current directory",
    "errortype": "*errors.errorString",
    "success": false,
    "tty": true,
    "version": "odo v3.0.0-rc2 (2f95b0242)",
    "cmdProperties": {
      "flags": [
        "devfile",
        "name",
        "starter",
        "v"
      ],
      "interactive": false,
      "isTelemetryEnabled": true
    }
  }
}
```

Reply:
Using slice was the first thing that I did. In Segment, everything was OK. But in Whoopra it ended up as `0=debug&1=no-watch`. Woopra properties can be only basic types (string, number, boolean) :-(

Reply:
Ah, too bad we can't pass the slice :-(
Okay then - thanks for the information.